jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java

```diff
@@ -135,6 +135,13 @@ private XADataSource createXaDataSource() {
             XADataSource xaDataSource = (XADataSource) Class.forName(className).newInstance();
             String url = driverProperties.getProperty("url", driverProperties.getProperty("URL"));
 
+            if (isH2()) {
+                final String username = driverProperties.getProperty("user");
+                final String password = driverProperties.getProperty("password");
+                xaDataSource.getClass().getMethod("setPassword", new Class[]{String.class}).invoke(xaDataSource, password);
+                xaDataSource.getClass().getMethod("setUser", new Class[]{String.class}).invoke(xaDataSource, username);
+            }
+
             if (!(className.startsWith("com.ibm.db2") || className.startsWith("com.sybase"))) {
```

Comment:
Shouldn't this be rather `else if` then?

Reply:
There is already `else` branch on line https://github.com/rsynek/jbpm/blob/b5feebfc6fce286b5e66ef07b6fb80963bf0ec0b/jbpm-test-util/src/main/java/org/jbpm/test/util/PoolingDataSource.java#L156 being currently executed for H2 too. I will think about making this part more readable - right now I wanted to see how many failures this fix eliminates.

Reply:
Ok, that's fair.

Reply:
A bit of refactoring we talked about - in the target (not yet used) component https://github.com/web-servers/narayana-tomcat/pull/1